CRM/Contact/Form/Contact.php

```diff
@@ -1192,47 +1192,22 @@ public function postProcess() {
    * @return bool
    *   true if data exists, false otherwise
    */
-  public static function blockDataExists(&$fields) {
+  public static function blockDataExists($fields): bool {
     if (!is_array($fields)) {
+      CRM_Core_Error::deprecatedWarning('support for invalid values will be dropped');
       return FALSE;
     }
 
-    static $skipFields = [
+    $dataFields = array_filter(array_diff_key($fields, array_fill_keys([
```

Comment:
Maybe it's just me, but I'm finding this hard to read with 3 array transformation functions all being called on the same line.

Reply:
hmm - yeah I'm just so used to that combo I guess - basically filter the diffed array to get the fields that are values but not the list to skip and are non-empty

Reply:
Ok not a blocker